### ISSUE TITLE
Can't access initWithQuery on FirebaseTableViewDataSource

### ISSUE THREAD

**Issue by muirandy** [firebase/FirebaseUI-iOS#62]
Just upgrading to the new release of Firebase (3.2.0) along with 0.4 of FirebaseUI. I'm using swift for my project.

Here's my podfile:
```
platform :ios, '8.0'
use_frameworks!
pod 'Firebase'
pod 'FirebaseUI', '~> 0.4'
```

It looks like the import mechanism has changed, from including FirebaseUI in the Bridging Header like so:
```
import <FirebaseUI/FirebaseDataSource.h>
import <FirebaseUI/FirebaseTableViewDataSource.h>
```
to using an include in the swift file, like so:
`import FirebaseUI.FirebaseTableViewDataSource`

I've run pod update (and pod install), and I am picking up the new Firebase classes ok. However, where I'm using FirebaseUI, I'm seeing this:

> Type FirebaseTableViewDataSource has no member 'initWithQuery'

Here's my code:
`self.dataSource = FirebaseTableViewDataSource.initWithQuery(query, prototypeReuseIdentifier: InstancesTableViewController.CELL_IDENTIFIER, view: self.tableView)
`
Now, it looks like initWithQuery is an instance method, rather than a class method. How can I instantiate a FirebaseTableViewDataSource so I can call the method?

Also, when I open FirebaseTableViewDataSource.h (by navigating through _Pods/FirebaseUI/Database/Frameworks/FirebaseDatabaseUI.framework/Headers_), I can see what appears to be an updated file (the header included a copyright by Google), and it has the methods I'm expecting. However, when I Cmd-Click on the use of FirebaseTableViewDataSource in my class (the code sample above), I see a different version of the class which appears to be older, and includes a method with the following signature:
`public init(query: FQuery, cellReuseIdentifier identifier: String, view tableView: UITableView)`

Any pointers? 

Andy

**Comment by JackP78** [comment]
This one seems to have been answered on StackOverflow (after much searching).  http://stackoverflow.com/questions/37311571/firebaseui-pod-install-error

Answer is you need you podfile to look like:
pod 'FirebaseUI', '~> 0.4'

And then wherever you previously just imported FirebaseUI, need to have the follow 2 imports:
import FirebaseUI
import FirebaseDatabaseUI

As noted on the stack overflow response they have changed the module names.  Eventually worked for me

**Comment by muirandy** [comment]
I'm afraid thats not the answer :(

It doesn't seem to be that it can't find FirebaseUI, its that it seems to be either the wrong version, or its picking up 2 versions.

I did try removing Firebase from the Podfile, just leaving FirebaseUI - but no effect. I also used the imports as suggested.


```
let query:FIRDatabaseQuery = ref.queryOrderedByChild("name")
self.dataSource = FirebaseTableViewDataSource.initWithQuery(query, prototypeReuseIdentifier: "myCellIdentifier", view: self.tableView)

```

> Type FirebaseTableViewDataSource has no member 'initWithQuery'

So where is it getting a different API from? Why does it think the method available is (old API):

`public init(query: FQuery, cellReuseIdentifier identifier: String, view tableView: UITableView)`

rather than (new API):

```
- (instancetype)initWithQuery:(FIRDatabaseQuery *)query
                     cellReuseIdentifier:(NSString *)identifier
                                    view:(UITableView *)tableView;

```
(and why does that look like an instance method rather than a class method??)

I'm stumped here - lost 2 days to this problem so far! 

Andy

**Comment by JackP78** [comment]
Well in swift the constructor is init, not initWithQuery.  In swift all initialisers are called init whereas objective C had the form initWithWhatever.  The signature is init(query: FIRDatabaseQuery, cellReuseIdentifer: String, view: UITableView)  NOT initWithQuery,  the latter is only for objective C.  However in order to see the first parameter as a FIRDatabaseQuery and not the old FQuery you need to import FirebaseDatabaseUI.

**Comment by muirandy** [comment]
Thanks for the help - its very much appreciated. I just wish I could say it was working.

My imports are:
```
import UIKit
import FirebaseDatabase
import FirebaseDatabaseUI
import FirebaseUI
import Firebase

```
My code is:
```
let query:FIRDatabaseQuery = firebaseTypesRef.queryOrderedByChild("name")
self.dataSource = FirebaseTableViewDataSource(query, cellReuseIdentifier:"MyCell", view: self.tableView)

```
And it doesn't work. I've tried prototypeReuseIdentifier rather than cellReuseIdentifier, no difference. It still seems to think it should be using the old FQuery (when I Cmd-Click on FirebaseTableViewDataSource).

I've tried taking out the imports 1 at a time, and it only breaks other lines of code.

I've also removed references from my bridging-header.

Yours lost,

Andy
